api-tests/dev_apis/crypto/test_c021/test_data.h

```diff
@@ -132,7 +132,7 @@ static const test_data check1[] = {
  PSA_KEY_DERIVATION_INPUT_SECRET, 256, {0}, 0,
  PSA_KEY_TYPE_RSA_KEY_PAIR, 2048, PSA_KEY_USAGE_EXPORT,
  PSA_ALG_RSA_PKCS1V15_SIGN_RAW,
- PSA_ERROR_INVALID_ARGUMENT
+ PSA_ERROR_NOT_SUPPORTED
```

Comment:
The v1.1.0 specification provides significant clarifications for error code usage. See https://armmbed.github.io/mbed-crypto/html/api/ops/kdf.html#c.psa_key_derivation_output_key for this particular API.

To enable efficient implementations, an implementation is permitted to return `PSA_ERROR_NOT_SUPPORTED` for valid key attributes that it does not support.

This is a general rule for most APIs in PSA Crypto. So most of the situations in which `PSA_ERROR_ARGUMENT_INVALID` is a valid response, could instead result in `PSA_ERROR_NOT_SUPPORTED` for some implementations.

Reply:
Yes:

- If the implementation has no support for the key type being tested, then it is permitted to return `PSA_ERROR_NOT_SUPPORTED`, rather than include the logic to check that the key size is valid for the key type.
- If the implementation does support the key type, and detects that this is an invalid key size for the type, it is recommended to return `PSA_ERROR_INVALID_ARGUMENT`.

Admittedly, if the key type being output by this function is not supported by the implementation, then the test case perhaps should not be executed at all?

Reply:
@athoelke agreed, we will make the necessary changes.